File: src/test/java/com/illuzor/afo/utils/ModulesUtilTest.java
package com.illuzor.afo.utils;

import org.junit.Test;
import org.junit.jupiter.api.Assertions;

import java.io.File;
import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.List;

public class ModulesUtilTest {

    @Test
    public void readFileToString_test() throws IOException {
        final String testString = "some string for test";
        File stringFile = new File("test.file");

        Files.write(stringFile.toPath(), testString.getBytes(Charset.forName("UTF-8")));
        Assertions.assertEquals(testString, ModulesUtil.readFileToString(stringFile));
        stringFile.delete();
    }

    @Test
    public void readFileToString_noFile_test() {
        Assertions.assertThrows(IOException.class, () -> ModulesUtil.readFileToString(new File("test1.file")));
    }

    @Test
    public void getModulesFromString_test() {
        String modulesString = "include ':app' ':storage' ':net' ':modules:module' ':modules:submodules:module'";
        String wrongString = "sdfs ser3 r3 sefkselrk";
        List<String> modulesList = new ArrayList<>(3);
        modulesList.add("app");
        modulesList.add("storage");
        modulesList.add("net");
        modulesList.add("modules/module");
        modulesList.add("modules/submodules/module");

        Assertions.assertEquals(modulesList, ModulesUtil.getModulesFromString(modulesString));
        Assertions.assertEquals(new ArrayList<String>(0), ModulesUtil.getModulesFromString(""));
        Assertions.assertEquals(new ArrayList<String>(0), ModulesUtil.getModulesFromString(wrongString));
    }
}

File: src/test/kotlin/com/illuzor/afo/utils/ModulesUtilTest.kt
package com.illuzor.afo.utils

import com.illuzor.afo.utils.ModulesUtil.getModulesFromString
import com.illuzor.afo.utils.ModulesUtil.readFileToString
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.io.File
import java.io.IOException
import java.nio.charset.Charset
import java.nio.file.Files

class ModulesUtilTest {

    @Test
    @Throws(IOException::class)
    fun readFileToString_test() {
        val testString = "some string for test"
        val stringFile = File("test.file")
        Files.write(stringFile.toPath(), testString.toByteArray(Charset.forName("UTF-8")))
        assertEquals(testString, readFileToString(stringFile))
        stringFile.delete()
    }

    @Test
    fun readFileToString_noFile_test() {
        assertThrows(IOException::class.java) { readFileToString(File("test1.file")) }
    }

    @Test
    fun modulesFromString_test (){
        val modulesString = "include ':app' ':storage' ':net' ':modules:module' ':modules:submodules:module'"
        val wrongString = "sdfs ser3 r3 sefkselrk"
        val modulesList: MutableList<String> = ArrayList(3)
        modulesList.add("app")
        modulesList.add("storage")
        modulesList.add("net")
        modulesList.add("modules/module")
        modulesList.add("modules/submodules/module")
        assertEquals(modulesList, getModulesFromString(modulesString))
        assertEquals(ArrayList<String>(0), getModulesFromString(""))
        assertEquals(ArrayList<String>(0), getModulesFromString(wrongString))
    }
}